packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -364,6 +365,50 @@ export function uniq(...args) {
 }
 
 /**
+  A computed property which returns a new array with all the unique
+  elements from an array, with uniqueness determined by specific key.
+  Example
+  ```javascript
+  var Hamster = Ember.Object.extend({
+    uniqueFruits: Ember.computed.uniqBy('fruits', 'id')
+  });
+  var hamster = Hamster.create({
+    fruits: [
+      { id: 1, 'banana' },
+      { id: 2, 'grape' },
+      { id: 3, 'peach' },
+      { id: 1, 'banana' }
+    ]
+  });
+  hamster.get('uniqueFruits'); // [ { id: 1, 'banana' }, { id: 2, 'grape' }, { id: 3, 'peach' }]
+  ```
+  @method uniqBy
+  @for Ember.computed
+  @param {String} dependentKey
+  @param {String} propertyKey
+  @return {Ember.ComputedProperty} computes a new array with all the
+  unique elements from the dependent array
+  @public
+*/
+export function uniqBy(dependentKey, propertyKey) {
+  return computed(`${dependentKey}.[]`, function() {
+    var uniq = emberA();
+    var seen = {};
```

Comment:
Wasn't @mmun asking that this use `EmptyObject` instead of `{}`? Can we do that?